### ISSUE TITLE
--no-native-fs fails on macOS Big Sur

### ISSUE THREAD

**Issue by porsager** [mpv-player/mpv#8490]
### Important Information

- mpv version 0.33.0
- macOS 11.1 (20C69)
- installed using `brew install mpv`

### Reproduction steps

Play a video with `--fullscreen` and `--no-native-fs`

### Expected behavior

The video will play in fullscreen on the same desktop

### Actual behavior

The video shows it's first frame and mpv crashes with the below error
```
video $ mpv a.mp4 --no-native-fs --fullscreen                                                                                                                                                                                      9ms 
2021-01-20 16:00:58.678 mpv[22417:624074] This NSLayoutConstraint is being configured with a constant that exceeds internal limits.  A smaller value will be substituted, but this problem should be fixed. Break on BOOL _NSLayoutConstraintNumberExceedsLimit(void) to debug.  This will be logged only once.  This may break in the future.
 (+) Video --vid=1 (*) (h264 1920x1080 25.000fps)
VO: [libmpv] 1920x1080 yuv420p
2021-01-20 16:00:59.062 mpv[22417:624074] NSWindowStyleMaskFullScreen set on a window outside of a full screen transition. Called from (
	0   AppKit                              0x00007fff22cb221c __25-[NSWindow setStyleMask:]_block_invoke + 116
	1   AppKit                              0x00007fff22cb214e NSPerformVisuallyAtomicChange + 132
	2   AppKit                              0x00007fff22cb205b -[NSWindow setStyleMask:] + 170
	3   mpv                                 0x0000000107220ced $s11macOS_swift6WindowC9styleMaskSo013NSWindowStyleF0VvsTo + 157
	4   mpv                                 0x000000010722239d $s11macOS_swift6WindowC15setToFullScreenyyF + 109
	5   mpv                                 0x0000000107221245 $s11macOS_swift6WindowC16toggleFullScreenyyypSgFTo + 101
	6   mpv                                 0x000000010721ce6e $sIeg_IeyB_TR + 14
	7   libdispatch.dylib                   0x00007fff201a05dd _dispatch_call_block_and_release + 12
	8   libdispatch.dylib                   0x00007fff201a17c7 _dispatch_client_callout + 8
	9   libdispatch.dylib                   0x00007fff201adb86 _dispatch_main_queue_callback_4CF + 940
	10  CoreFoundation                      0x00007fff20481970 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	11  CoreFoundation                      0x00007fff20443852 __CFRunLoopRun + 2731
	12  CoreFoundation                      0x00007fff204426ce CFRunLoopRunSpecific + 563
	13  HIToolbox                           0x00007fff286c76d0 RunCurrentEventLoopInMode + 292
	14  HIToolbox                           0x00007fff286c74cc ReceiveNextEventCommon + 709
	15  HIToolbox                           0x00007fff286c71ef _BlockUntilNextEventMatchingListInModeWithFilter + 64
	16  AppKit                              0x00007fff22c5fde9 _DPSNextEvent + 883
	17  AppKit                              0x00007fff22c5e5af -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1366
	18  AppKit                              0x00007fff22c50b0a -[NSApplication run] + 586
	19  mpv                                 0x000000010720de04 cocoa_main + 756
	20  libdyld.dylib                       0x00007fff20367621 start + 1
)
2021-01-20 16:00:59.065 mpv[22417:624074] *** Terminating app due to uncaught exception 'NSGenericException', reason: 'NSWindowStyleMaskFullScreen set on a window outside of a full screen transition. Called from (
	0   AppKit                              0x00007fff22cb221c __25-[NSWindow setStyleMask:]_block_invoke + 116
	1   AppKit                              0x00007fff22cb214e NSPerformVisuallyAtomicChange + 132
	2   AppKit                              0x00007fff22cb205b -[NSWindow setStyleMask:] + 170
	3   mpv                                 0x0000000107220ced $s11macOS_swift6WindowC9styleMaskSo013NSWindowStyleF0VvsTo + 157
	4   mpv                                 0x000000010722239d $s11macOS_swift6WindowC15setToFullScreenyyF + 109
	5   mpv                                 0x0000000107221245 $s11macOS_swift6WindowC16toggleFullScreenyyypSgFTo + 101
	6   mpv                                 0x000000010721ce6e $sIeg_IeyB_TR + 14
	7   libdispatch.dylib                   0x00007fff201a05dd _dispatch_call_block_and_release + 12
	8   libdispatch.dylib                   0x00007fff201a17c7 _dispatch_client_callout + 8
	9   libdispatch.dylib                   0x00007fff201adb86 _dispatch_main_queue_callback_4CF + 940
	10  CoreFoundation                      0x00007fff20481970 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	11  CoreFoundation                      0x00007fff20443852 __CFRunLoopRun + 2731
	12  CoreFoundation                      0x00007fff204426ce CFRunLoopRunSpecific + 563
	13  HIToolbox                           0x00007fff286c76d0 RunCurrentEventLoopInMode + 292
	14  HIToolbox                           0x00007fff286c74cc ReceiveNextEventCommon + 709
	15  HIToolbox                           0x00007fff286c71ef _BlockUntilNextEventMatchingListInModeWithFilter + 64
	16  AppKit                              0x00007fff22c5fde9 _DPSNextEvent + 883
	17  AppKit                              0x00007fff22c5e5af -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1366
	18  AppKit                              0x00007fff22c50b0a -[NSApplication run] + 586
	19  mpv                                 0x000000010720de04 cocoa_main + 756
	20  libdyld.dylib                       0x00007fff20367621 start + 1
)'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff204be6af __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00007fff201f63c9 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff204be513 +[NSException raise:format:] + 189
	3   AppKit                              0x00007fff22cb2259 __25-[NSWindow setStyleMask:]_block_invoke + 177
	4   AppKit                              0x00007fff22cb214e NSPerformVisuallyAtomicChange + 132
	5   AppKit                              0x00007fff22cb205b -[NSWindow setStyleMask:] + 170
	6   mpv                                 0x0000000107220ced $s11macOS_swift6WindowC9styleMaskSo013NSWindowStyleF0VvsTo + 157
	7   mpv                                 0x000000010722239d $s11macOS_swift6WindowC15setToFullScreenyyF + 109
	8   mpv                                 0x0000000107221245 $s11macOS_swift6WindowC16toggleFullScreenyyypSgFTo + 101
	9   mpv                                 0x000000010721ce6e $sIeg_IeyB_TR + 14
	10  libdispatch.dylib                   0x00007fff201a05dd _dispatch_call_block_and_release + 12
	11  libdispatch.dylib                   0x00007fff201a17c7 _dispatch_client_callout + 8
	12  libdispatch.dylib                   0x00007fff201adb86 _dispatch_main_queue_callback_4CF + 940
	13  CoreFoundation                      0x00007fff20481970 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	14  CoreFoundation                      0x00007fff20443852 __CFRunLoopRun + 2731
	15  CoreFoundation                      0x00007fff204426ce CFRunLoopRunSpecific + 563
	16  HIToolbox                           0x00007fff286c76d0 RunCurrentEventLoopInMode + 292
	17  HIToolbox                           0x00007fff286c74cc ReceiveNextEventCommon + 709
	18  HIToolbox                           0x00007fff286c71ef _BlockUntilNextEventMatchingListInModeWithFilter + 64
	19  AppKit                              0x00007fff22c5fde9 _DPSNextEvent + 883
	20  AppKit                              0x00007fff22c5e5af -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1366
	21  AppKit                              0x00007fff22c50b0a -[NSApplication run] + 586
	22  mpv                                 0x000000010720de04 cocoa_main + 756
	23  libdyld.dylib                       0x00007fff20367621 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
zsh: abort      mpv a.mp4 --no-native-fs --fullscreen
```

### Log file

[output.txt](https://github.com/mpv-player/mpv/files/5843329/output.txt)

### Sample files

Any video file I've tried does this.

**Comment by Akemi** [comment]
so it's time to remove the none native fullscreen.

**Comment by porsager** [comment]
Do you recommend using geometry to achieve the same functionality then?

**Comment by Akemi** [comment]
i recommend using it not at all. it's a semi deprecated broken in many ways of doing fullscreen.

i was half joking about removing it. it seems easy to fix, hopefully. i will only remove it, if it makes too much trouble maintaining it.

using `--geometry` will most likely not work because of the geometry calculation that takes the menubar into account. no way of avoiding the rounded corners atm. i tired to add functionality for it, but it caused other problems again.

**Comment by porsager** [comment]
Ok, cool

It's a pity they are so keen on forcing their rounded corners and borders on apps. I'll manage for now with hiding the menubar and using `--geometry`, but I hope you're right that the fix is easy 🙂

**Comment by Akemi** [comment]
just to confirm, does this only happen on startup when using --fullscreen? or does it also crash when starting mpv windowed and using 'f' or double click to go fullscreen?

asking since #8507 seems to be the same underlying issue but without a crash. which would be really weird.

**Comment by porsager** [comment]
It also happens if pressing `f` during playback

**Comment by aerobounce** [comment]
I'm on Big Sur and it happens on my env too.
`11.2.1（20D74）`
`mpv 0.33.0`

Did some tests: ` mpv --no-config --no-native-fs 'some-video.mp4'`

- Double click video rectangle
  - Crash
- Press `F` Key
  - Crash
- Click Maximize button in title bar
  - The button causes stacktrace, but mpv is kept functional. Then when mpv quits, mpv Freezes. You have to manually kill the process.

With all the 3 cases above, stacktraces are basically the same:
```
2021-02-11 16:45:25.646 mpv[39222:232036] NSWindowStyleMaskFullScreen set on a window outside of a full screen transition. Called from (
        0   AppKit                              0x00007fff22cdcddc __25-[NSWindow setStyleMask:]_block_invoke + 116
        1   AppKit                              0x00007fff22cdcd0e NSPerformVisuallyAtomicChange + 132
        2   AppKit                              0x00007fff22cdcc1b -[NSWindow setStyleMask:] + 170
        3   mpv                                 0x000000010227dcdd $s11macOS_swift6WindowC9styleMaskSo013NSWindowStyleF0VvsTo + 157
        4   mpv                                 0x000000010227f38d $s11macOS_swift6WindowC15setToFullScreenyyF + 109
        5   mpv                                 0x000000010227e235 $s11macOS_swift6WindowC16toggleFullScreenyyypSgFTo + 101
```

---

However, I found this option as an alternate: `macos-fs-animation-duration=0`
It just works and I'm happy.

**Comment by Akemi** [comment]
i made a PR #8596 with a possible fix. it would be nice if you could test it.

**Comment by porsager** [comment]
Beautiful 👏 Just tried it out now - it works!

Thank you !

**Comment by Akemi** [comment]
thx for testing.

**Comment by porsager** [comment]
Is there any chance this can be merged into master?

**Comment by porsager** [comment]
Hmm.. sorry, it seems like it's merged..

This is the error I'm seeing with 0.33.1
```
lib $ mpv test.mp4 --fullscreen --no-native-fs --v                                                                                                                                                                        115ms 
[cplayer] Command line options: 'test.mp4' '--fullscreen' '--no-native-fs' '--v'
[cplayer] mpv 0.33.1 Copyright © 2000-2020 mpv/MPlayer/mplayer2 projects
[cplayer]  built on Mon Apr  5 18:52:46 BST 2021
[cplayer] FFmpeg library versions:
[cplayer]    libavutil       56.51.100 (runtime 56.70.100)
[cplayer]    libavcodec      58.91.100 (runtime 58.134.100)
[cplayer]    libavformat     58.45.100 (runtime 58.76.100)
[cplayer]    libswscale      5.7.100 (runtime 5.9.100)
[cplayer]    libavfilter     7.85.100 (runtime 7.110.100)
[cplayer]    libswresample   3.7.100 (runtime 3.9.100)
[cplayer] FFmpeg version: 4.4
[cplayer] 
[cplayer] Configuration: waf configure --prefix=/usr/local/Cellar/mpv/0.33.1 --enable-html-build --enable-javascript --enable-libmpv-shared --enable-lua --enable-libarchive --enable-uchardet --confdir=/usr/local/etc/mpv --datadir=/usr/local/Cellar/mpv/0.33.1/share/mpv --mandir=/usr/local/Cellar/mpv/0.33.1/share/man --docdir=/usr/local/Cellar/mpv/0.33.1/share/doc/mpv --zshdir=/usr/local/Cellar/mpv/0.33.1/share/zsh/site-functions --lua=luajit
[cplayer] List of enabled features: asm bsd-fstatfs build-date cocoa coreaudio cplayer cplugins debug-build ffmpeg gl gl-cocoa glob glob-posix gpl iconv javascript jpeg lcms2 libarchive libass libavdevice libbluray libdl libm libmpv-shared lua luajit macos-10-11-features macos-10-12-2-features macos-10-14-features macos-cocoa-cb macos-media-player macos-touchbar optimize osx-thread-name plain-gl posix posix-or-mingw pthreads rubberband stdatomic swift uchardet vapoursynth videotoolbox-gl videotoolbox-hwaccel zimg zlib
[cplayer] Reading config file /usr/local/etc/mpv/encoding-profiles.conf
[cplayer] Applying profile 'default'...
[cplayer] Setting option 'fullscreen' = '' (flags = 8)
[cplayer] Setting option 'native-fs' = 'no' (flags = 8)
[cplayer] Setting option 'v' = '' (flags = 8)
2021-09-07 20:38:46.663 mpv[85590:3916430] This NSLayoutConstraint is being configured with a constant that exceeds internal limits.  A smaller value will be substituted, but this problem should be fixed. Break on BOOL _NSLayoutConstraintNumberExceedsLimit(void) to debug.  This will be logged only once.  This may break in the future.
[osx/cocoacb] Created CGL pixel format with attributes: kCGLPFAOpenGLProfile, kCGLOGLPVersion_3_2_Core, kCGLPFAAccelerated, kCGLPFADoubleBuffer, kCGLPFAColorSize, 64, kCGLPFAColorFloat, kCGLPFABackingStore, kCGLPFAAllowOfflineRenderers, kCGLPFASupportsAutomaticGraphicsSwitching, 0
[libmpv_render] GL_VERSION='4.1 ATI-4.6.20'
[libmpv_render] Detected desktop OpenGL 4.1.
[libmpv_render] GL_VENDOR='ATI Technologies Inc.'
[libmpv_render] GL_RENDERER='AMD Radeon Pro 560 OpenGL Engine'
[libmpv_render] GL_SHADING_LANGUAGE_VERSION='4.10'
[libmpv_render] Loaded extension GL_APPLE_rgb_422.
[libmpv_render] GL_*_swap_control extension missing.
[libmpv_render] Testing FBO format rgba16f
[libmpv_render] Using FBO format rgba16f.
[libmpv_render] Disabling HDR peak computation (one or more of the following is not supported: compute shaders=0, SSBO=0).
[libmpv_render] No advanced processing required. Enabling dumb mode.
[libmpv_render] Loading hwdec driver 'videotoolbox'
[cplayer] Waiting for scripts...
[osd/libass] libass API version: 0x1501000
[osd/libass] libass source: tarball: 0.15.1
[osd/libass] Shaper: FriBidi 1.0.10 (SIMPLE) HarfBuzz-ng 2.8.2 (COMPLEX)
[osd/libass] Setting up fonts...
[osd/libass] Using font provider coretext
[osd/libass] Done.
[cplayer] Set property: shared-script-properties -> 1
[cplayer] Done loading scripts.
[cplayer] Running hook: ytdl_hook/on_load
[ytdl_hook] ytdl:// hook 
[ytdl_hook] not a ytdl:// url 
[bdmv/bluray] Opening test.mp4
[file] Opening test.mp4
[demux] Trying demuxers for level=normal.
[cplayer] Set property: shared-script-properties -> 1
[osd/libass] libass API version: 0x1501000
[osd/libass] libass source: tarball: 0.15.1
[osd/libass] Shaper: FriBidi 1.0.10 (SIMPLE) HarfBuzz-ng 2.8.2 (COMPLEX)
[osd/libass] Setting up fonts...
[osd/libass] Using font provider coretext
[osd/libass] Done.
[cplayer] Set property: shared-script-properties -> 1
[lavf] Found 'mov,mp4,m4a,3gp,3g2,mj2' at score=100 size=2048.
[file] stream level seek from 131072 to 303098
[demux] Detected file format: mov,mp4,m4a,3gp,3g2,mj2 (libavformat)
[cplayer] Opening done: test.mp4
[find_files] Loading external files in .
[cplayer] Running hook: ytdl_hook/on_preloaded
[lavf] select track 0
[cplayer]  (+) Video --vid=1 (*) (h264 1920x1080 25.000fps)
[osx/cocoacb] Can't find an ambient light sensor
[vo/libmpv] Assuming 60.000877 FPS for display sync.
[vd] Container reported FPS: 25.000000
[vd] Codec list:
[vd]     h264 - H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
[vd] Opening decoder h264
[vd] No hardware decoding requested.
[vd] Using software decoding.
[vd] Detected 8 logical cores.
[vd] Requesting 9 threads for decoding.
[vd] Selected codec: h264 (H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10)
[vf] User filter list:
[vf]   (empty)
[cplayer] Starting playback...
[vd] DR failed - disabling.
[vd] Using software decoding.
[vd] Decoder format: 1920x1080 [0:1] yuv420p bt.709/bt.709/bt.1886/limited/auto CL=mpeg2/4/h264
[vf] [in] 1920x1080 yuv420p bt.709/bt.709/bt.1886/limited/display SP=1.000000 CL=mpeg2/4/h264
[vf] [userdeint] 1920x1080 yuv420p bt.709/bt.709/bt.1886/limited/display SP=1.000000 CL=mpeg2/4/h264
[vf] [userdeint] (disabled)
[vf] [autorotate] 1920x1080 yuv420p bt.709/bt.709/bt.1886/limited/display SP=1.000000 CL=mpeg2/4/h264
[vf] [autorotate] (disabled)
[vf] [convert] 1920x1080 yuv420p bt.709/bt.709/bt.1886/limited/display SP=1.000000 CL=mpeg2/4/h264
[vf] [convert] (disabled)
[vf] [out] 1920x1080 yuv420p bt.709/bt.709/bt.1886/limited/display SP=1.000000 CL=mpeg2/4/h264
[cplayer] VO: [libmpv] 1920x1080 yuv420p
[cplayer] VO: Description: render API for libmpv
[vo/libmpv] reconfig to 1920x1080 yuv420p bt.709/bt.709/bt.1886/limited/display SP=1.000000 CL=mpeg2/4/h264
[libmpv_render] Texture for plane 0: 1920x1080
[libmpv_render] Texture for plane 1: 960x540
[libmpv_render] Texture for plane 2: 960x540
[libmpv_render] Testing FBO format rgba16f
[libmpv_render] Using FBO format rgba16f.
[libmpv_render] No advanced processing required. Enabling dumb mode.
[cplayer] first video frame after restart shown
[cplayer] playback restart complete @ 0.000000, audio=eof, video=playing
[statusline] V: 00:00:00 / 00:01:00 (0%)
[cplayer] Set property: shared-script-properties -> 1
[libmpv_render] Window size: 1920x1080 (Borders: l=0 t=0 r=0 b=0)
[libmpv_render] Video source: 1920x1080 (1:1)
[libmpv_render] Video display: (0, 0) 1920x1080 -> (0, 0) 1920x1080
[libmpv_render] Video scale: 1.000000/1.000000
[libmpv_render] OSD borders: l=0 t=0 r=0 b=0
[libmpv_render] Video borders: l=0 t=0 r=0 b=0
[cplayer] Set property: shared-script-properties -> 1
[statusline] V: 00:00:00 / 00:01:00 (0%)
[libmpv_render] Window size: 3840x2160 (Borders: l=0 t=0 r=0 b=0)
[libmpv_render] Video source: 1920x1080 (1:1)
[libmpv_render] Video display: (0, 0) 1920x1080 -> (0, 0) 3840x2160
[libmpv_render] Video scale: 2.000000/2.000000
[libmpv_render] OSD borders: l=0 t=0 r=0 b=0
[libmpv_render] Video borders: l=0 t=0 r=0 b=0
[cplayer] Set property: shared-script-properties -> 1
2021-09-07 20:38:47.236 mpv[85590:3916430] NSWindowStyleMaskFullScreen set on a window outside of a full screen transition. Called from (
	0   AppKit                              0x00007fff22d38826 __25-[NSWindow setStyleMask:]_block_invoke + 125
	1   AppKit                              0x00007fff22d3874f NSPerformVisuallyAtomicChange + 132
	2   AppKit                              0x00007fff22d3865c -[NSWindow setStyleMask:] + 170
	3   mpv                                 0x000000010e43dcdd $s11macOS_swift6WindowC9styleMaskSo013NSWindowStyleF0VvsTo + 157
	4   mpv                                 0x000000010e43f38d $s11macOS_swift6WindowC15setToFullScreenyyF + 109
	5   mpv                                 0x000000010e43e235 $s11macOS_swift6WindowC16toggleFullScreenyyypSgFTo + 101
	6   mpv                                 0x000000010e439e5e $sIeg_IeyB_TR + 14
	7   libdispatch.dylib                   0x00007fff2023c623 _dispatch_call_block_and_release + 12
	8   libdispatch.dylib                   0x00007fff2023d806 _dispatch_client_callout + 8
	9   libdispatch.dylib                   0x00007fff20249b4f _dispatch_main_queue_callback_4CF + 940
	10  CoreFoundation                      0x00007fff2051d5a8 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	11  CoreFoundation                      0x00007fff204df7a2 __CFRunLoopRun + 2755
	12  CoreFoundation                      0x00007fff204de61c CFRunLoopRunSpecific + 563
	13  HIToolbox                           0x00007fff28724a83 RunCurrentEventLoopInMode + 292
	14  HIToolbox                           0x00007fff287247e5 ReceiveNextEventCommon + 587
	15  HIToolbox                           0x00007fff28724583 _BlockUntilNextEventMatchingListInModeWithFilter + 70
	16  AppKit                              0x00007fff22ce6172 _DPSNextEvent + 864
	17  AppKit                              0x00007fff22ce4945 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1364
	18  AppKit                              0x00007fff22cd6c69 -[NSApplication run] + 586
	19  mpv                                 0x000000010e42adf4 cocoa_main + 756
	20  libdyld.dylib                       0x00007fff20402f3d start + 1
	21  ???                                 0x0000000000000005 0x0 + 5
)
2021-09-07 20:38:47.238 mpv[85590:3916430] *** Terminating app due to uncaught exception 'NSGenericException', reason: 'NSWindowStyleMaskFullScreen set on a window outside of a full screen transition. Called from (
	0   AppKit                              0x00007fff22d38826 __25-[NSWindow setStyleMask:]_block_invoke + 125
	1   AppKit                              0x00007fff22d3874f NSPerformVisuallyAtomicChange + 132
	2   AppKit                              0x00007fff22d3865c -[NSWindow setStyleMask:] + 170
	3   mpv                                 0x000000010e43dcdd $s11macOS_swift6WindowC9styleMaskSo013NSWindowStyleF0VvsTo + 157
	4   mpv                                 0x000000010e43f38d $s11macOS_swift6WindowC15setToFullScreenyyF + 109
	5   mpv                                 0x000000010e43e235 $s11macOS_swift6WindowC16toggleFullScreenyyypSgFTo + 101
	6   mpv                                 0x000000010e439e5e $sIeg_IeyB_TR + 14
	7   libdispatch.dylib                   0x00007fff2023c623 _dispatch_call_block_and_release + 12
	8   libdispatch.dylib                   0x00007fff2023d806 _dispatch_client_callout + 8
	9   libdispatch.dylib                   0x00007fff20249b4f _dispatch_main_queue_callback_4CF + 940
	10  CoreFoundation                      0x00007fff2051d5a8 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	11  CoreFoundation                      0x00007fff204df7a2 __CFRunLoopRun + 2755
	12  CoreFoundation                      0x00007fff204de61c CFRunLoopRunSpecific + 563
	13  HIToolbox                           0x00007fff28724a83 RunCurrentEventLoopInMode + 292
	14  HIToolbox                           0x00007fff287247e5 ReceiveNextEventCommon + 587
	15  HIToolbox                           0x00007fff28724583 _BlockUntilNextEventMatchingListInModeWithFilter + 70
	16  AppKit                              0x00007fff22ce6172 _DPSNextEvent + 864
	17  AppKit                              0x00007fff22ce4945 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1364
	18  AppKit                              0x00007fff22cd6c69 -[NSApplication run] + 586
	19  mpv                                 0x000000010e42adf4 cocoa_main + 756
	20  libdyld.dylib                       0x00007fff20402f3d start + 1
	21  ???                                 0x0000000000000005 0x0 + 5
)'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff2055a83b __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00007fff20292d92 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff2055a69f +[NSException raise:format:] + 189
	3   AppKit                              0x00007fff22d38863 __25-[NSWindow setStyleMask:]_block_invoke + 186
	4   AppKit                              0x00007fff22d3874f NSPerformVisuallyAtomicChange + 132
	5   AppKit                              0x00007fff22d3865c -[NSWindow setStyleMask:] + 170
	6   mpv                                 0x000000010e43dcdd $s11macOS_swift6WindowC9styleMaskSo013NSWindowStyleF0VvsTo + 157
	7   mpv                                 0x000000010e43f38d $s11macOS_swift6WindowC15setToFullScreenyyF + 109
	8   mpv                                 0x000000010e43e235 $s11macOS_swift6WindowC16toggleFullScreenyyypSgFTo + 101
	9   mpv                                 0x000000010e439e5e $sIeg_IeyB_TR + 14
	10  libdispatch.dylib                   0x00007fff2023c623 _dispatch_call_block_and_release + 12
	11  libdispatch.dylib                   0x00007fff2023d806 _dispatch_client_callout + 8
	12  libdispatch.dylib                   0x00007fff20249b4f _dispatch_main_queue_callback_4CF + 940
	13  CoreFoundation                      0x00007fff2051d5a8 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	14  CoreFoundation                      0x00007fff204df7a2 __CFRunLoopRun + 2755
	15  CoreFoundation                      0x00007fff204de61c CFRunLoopRunSpecific + 563
	16  HIToolbox                           0x00007fff28724a83 RunCurrentEventLoopInMode + 292
	17  HIToolbox                           0x00007fff287247e5 ReceiveNextEventCommon + 587
	18  HIToolbox                           0x00007fff28724583 _BlockUntilNextEventMatchingListInModeWithFilter + 70
	19  AppKit                              0x00007fff22ce6172 _DPSNextEvent + 864
	20  AppKit                              0x00007fff22ce4945 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1364
	21  AppKit                              0x00007fff22cd6c69 -[NSApplication run] + 586
	22  mpv                                 0x000000010e42adf4 cocoa_main + 756
	23  libdyld.dylib                       0x00007fff20402f3d start + 1
	24  ???                                 0x0000000000000005 0x0 + 5
)
libc++abi: terminating with uncaught exception of type NSException
zsh: abort      mpv test.mp4 --fullscreen --no-native-fs --v
```

**Comment by Akemi** [comment]
this fix is in master only and in no release.

**Comment by porsager** [comment]
Ah, thank you! sorry I couldn't figure that out.

**Comment by forthrin** [comment]
@Akemi: This issue, commited and resolved almost two years ago (February 2021), still happens for me with the newly released 0.35.0. How can this be? Or is this a regression?

**Comment by Akemi** [comment]
because the none native fullscreen ist deprecated by Apple and they give a shit about it and breaking it with newer OSes. if something like this breaks it's mostly Apples fault.

also who know if it's even the same issue without any info given.

please don't necro old issues.

**Comment by forthrin** [comment]
@porsager: Suggest to close this issue as it was reported for 0.33, and there are reports for newer versions.

**Comment by zohozer** [comment]
Ho I can add the "--no-native-fs" into the mpv.conf? Should be "no-native-fs=yes" ? If I pass the "--no-native-fs" into commandline it is working but if I put the "no-native-fs=yes" into mpv.conf it is not working.

**Comment by llyyr** [comment]
https://mpv.io/manual/master/#usage

> Every flag option has a no-flag counterpart, e.g. the opposite of the --fs option is --no-fs. --fs=yes is same as --fs, --fs=no is the same as --no-fs.

**Comment by zohozer** [comment]
```
fs=yes
native-fs=yes
```

still not working.

**Comment by Akemi** [comment]
please don't necro closed issues. if you have an issue open a new on with all infos we ask you to provide.

that config is wrong if you want `no-native-fs`. properly read the documentation that was linked. it's either `no-native-fs` or `native-fs=no`. the same as on the command line, just without the leading `--`
